lib/elixir_sense/providers/implementation.ex

```diff
@@ -0,0 +1,217 @@
+    |> Enum.map(fn implementation ->
+      Location.find_source({implementation, maybe_callback}, nil)
+    end)
+  end
+
+  defp is_callback(behaviour, fun) when is_atom(behaviour) do
+    Code.ensure_loaded?(behaviour) and
+      function_exported?(behaviour, :behaviour_info, 1) and
+      behaviour.behaviour_info(:callbacks)
+      |> Enum.any?(&match?({^fun, _}, &1))
+  end
+
+  defp find_delegatee(
+         {module, function},
+         mods_funs_to_positions,
+         env,
+         metadata_types,
+         binding_env,
+         visited \\ []
+       ) do
+    unless {module, function} in visited do
+      do_find_delegatee(
+        {module, function},
+        mods_funs_to_positions,
+        env,
+        metadata_types,
+        binding_env,
+        [{module, function} | visited]
+      )
+    end
+  end
+
+  # credo:disable-for-lines:20
+  defp do_find_delegatee(
+         {{:attribute, _attr} = type, function},
+         mods_funs_to_positions,
+         env,
+         metadata_types,
+         binding_env,
+         visited
+       ) do
+    case Binding.expand(binding_env, type) do
+      {:atom, module} ->
+        do_find_delegatee(
+          {Introspection.expand_alias(module, env.aliases), function},
+          mods_funs_to_positions,
+          env,
+          metadata_types,
+          binding_env,
+          visited
+        )
+
+      _ ->
+        nil
+    end
+  end
+
+  defp do_find_delegatee(
+         {module, function},
+         mods_funs_to_positions,
+         env,
+         metadata_types,
+         binding_env,
+         visited
+       ) do
+    %State.Env{
+      module: current_module,
+      imports: imports,
+      aliases: aliases
+    } = env
+
+    case {module, function}
+         |> Introspection.actual_mod_fun(
+           imports,
+           aliases,
+           current_module,
+           mods_funs_to_positions,
+           metadata_types
+         ) do
+      {mod, fun, true} when not is_nil(fun) ->
+        case mods_funs_to_positions[{mod, fun, nil}] do
+          nil ->
+            find_delegatee_location(mod, fun, current_module, visited)
+
+          %ModFunInfo{type: :defdelegate, target: target} when not is_nil(target) ->
+            find_delegatee(
+              target,
+              mods_funs_to_positions,
+              env,
+              metadata_types,
+              binding_env,
+              visited
+            )
+
+          _ ->
+            # not a delegate
+            nil
+        end
+
+      _ ->
+        nil
+    end
+  end
+
+  defp find_delegatee_location(mod, fun, current_module, visited) do
+    case Normalized.Code.get_docs(mod, :docs)
+         |> List.wrap()
+         |> Enum.find(&match?({{^fun, _}, _, :function, _, _, %{delegate_to: _}}, &1)) do
+      nil ->
+        # ensure we are expanding a delegate
+        if length(visited) > 1 do
+          Location.find_source({mod, fun}, current_module)
+        end
```

Comment:
Since this will be called multiple times, it might be worth it to use a more efficient version:

```elixir
case visited do
  # Length > 1
  [_, _ | _] -> 
    Location.find_source({mod, fun}, current_module)
  _ -> 
    nil
end
```

It looks uglier but it will prevent traversing the whole list every time `length/1` is called.